### PR TITLE
drivers: spi: spi_ll_stm32: cleanup

### PR DESCRIPTION
**Pull request by yboutreux**
Long line cleanup and explicit hard-coded values inside the driver.

DEPENDS ON : https://github.com/arnop2/zephyr_stm32mp1/pull/15

### CODE REVIEW

File: drivers/spi/spi_ll_stm32.c [hunk 1]
```diff
@@ -180,24 +246,52 @@ static void spi_stm32_complete(struct spi_stm32_data *data, SPI_TypeDef *spi,
...
+#endif	/* CONFIG_SOC_SERIES_STM32MP1X */
+#endif	/* CONFIG_SPI_STM32_HAS_FIFO */
 
 	if (LL_SPI_GetMode(spi) == LL_SPI_MODE_MASTER) {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+		while (LL_SPI_IsEnabled(spi) &&
+		       !LL_SPI_IsActiveFlag_MODF(spi) &&
+		       !LL_SPI_IsActiveFlag_TXC(spi)) {
+			/* NOP */
+		}
```
Comment (arnop2) [inline drivers/spi/spi_ll_stm32.c]: Please double check thes conditions. for me
LL_SPI_IsEnabled(spi) always true
!LL_SPI_IsActiveFlag_MODF(spi) always true 
!LL_SPI_IsActiveFlag_TXC(spi) to check if always false as we are in  spi_stm32_complete function...

Comment (yboutreux) [reply drivers/spi/spi_ll_stm32.c]: For context, BSY flag isn't in the STM32MP1 SPI registers.
Following STM32L4 BSY Flag definition : 
```
/**
  * @brief  Get busy flag
  * @note   The BSY flag is cleared under any one of the following conditions:
  * -When the SPI is correctly disabled
  * -When a fault is detected in Master mode (MODF bit set to 1)
  * -In Master mode, when it finishes a data transmission and no new data is ready to be
  * sent
  * -In Slave mode, when the BSY flag is set to '0' for at least one SPI clock cycle between
  * each data transfer.
  * @rmtoll SR           BSY           LL_SPI_IsActiveFlag_BSY
  * @param  SPIx SPI Instance
  * @retval State of bit (1 or 0).
  */
```

Comment (yboutreux) [reply drivers/spi/spi_ll_stm32.c]: Wrong PR to address this subject

Comment (yboutreux) [reply drivers/spi/spi_ll_stm32.c]: > Please double check thes conditions. for me
> LL_SPI_IsEnabled(spi) always true
> !LL_SPI_IsActiveFlag_MODF(spi) always true
> !LL_SPI_IsActiveFlag_TXC(spi) to check if always false as we are in spi_stm32_complete function...

Wrong PR to address this subject, please see [here](https://github.com/arnop2/zephyr_stm32mp1/pull/15/files#r306799157)

File: drivers/spi/spi_ll_stm32.c [hunk 2]
```diff
@@ -20,6 +20,17 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
 
 #include "spi_ll_stm32.h"
 
+/* Size of the SPI buffer */
+#define WORD_8_BITS	8
+#define WORD_16_BITS	16
```
Comment (arnop2) [inline drivers/spi/spi_ll_stm32.c]: could you not define it but just use sizeof(u8_t) and siezeof(u16_t)

File: drivers/spi/spi_ll_stm32.c [hunk 3]
```diff
@@ -20,6 +20,17 @@ LOG_MODULE_REGISTER(spi_ll_stm32);
...
+
+/* Size (in bytes) of a chunk */
+#define CHUNK_1_BYTE	1
+#define CHUNK_2_BYTES	2
```
Comment (arnop2) [inline drivers/spi/spi_ll_stm32.c]: #define  FRAME_SIZE(x) (sizeof(x) /sizeof(u8_t))

File: drivers/spi/spi_ll_stm32.c [hunk 4]
```diff
@@ -119,14 +131,14 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 	}
 #endif /* CONFIG_SOC_SERIES_STM32MP1X */
 
-	if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
+	if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == WORD_8_BITS) {
 		LL_SPI_TransmitData8(spi, tx_frame);
 		/* The update is ignored if TX is off. */
-		spi_context_update_tx(&data->ctx, 1, 1);
+		spi_context_update_tx(&data->ctx, CHUNK_1_BYTE, CHUNK_SIZE);
```
Comment (arnop2) [inline drivers/spi/spi_ll_stm32.c]: I would keep 1 and not define CHUNK_SIZE that looks for me confusing.
spi_context_update_tx(&data->ctx, FRAME_SIZE(u16_t), 1);

**Pull request closed without merging**